plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/cache/DeltaLakeCacheKeyProvider.java

```diff
@@ -27,8 +27,13 @@ public class DeltaLakeCacheKeyProvider
     @Override
     public Optional<String> getCacheKey(TrinoInputFile delegate)
     {
-        // TODO: Consider caching of the Parquet checkpoint files within _delta_log
-        if (!delegate.location().path().contains("/_delta_log/")) {
+        String path = delegate.location().path();
+        if (path.endsWith(".trinoSchema") || path.contains("/.trinoPermissions/")) {
+            // Needed to avoid caching files from FileHiveMetastore on coordinator during tests
+            return Optional.empty();
+        }
+        // Cache data files and checkpoint files within the delta log
```

Comment:
Could you add a comment here about why the json delta log entries can't be cached ?
My impression after reading https://github.com/delta-io/delta/issues/1975 was that everything except `_last_checkpoint` is immutable.

Reply:
`extended_stats.json` from Trino are also not immutable.
https://github.com/trinodb/trino/blob/2536dc2adbbf42fb2bf13e904b5899cb4ee66241/plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/statistics/MetaDirStatisticsAccess.java#L41-L45

Reply:
I'm hesitant to cache .json files because of the comments on ADLS adding the files and then writing the contents. We could make caching of commits configurable?

Reply:
>  because of the comments on ADLS adding the files and then writing the contents.

Which comments are those?

Reply:
Based on the comments at https://github.com/delta-io/delta/issues/1975#issuecomment-1734296398
I still get the impression that caching json delta log entries is valid. I wouldn't make it configurable, I think we should start by caching it in initial implementation and roll it back later if it's found to be problematic in practise.

Reply:
As for `extendeded_stats.json`, I think we could cache that with modification time included in the cache key. But that would still make HEAD call to S3 and these are small files, so it's unclear to me if a HEAD request would have a perf advantage over a small GET request. We can probably exclude stats json.